datalad_next/archive_operations/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""Archive operation handlers"""
+
+# allow for |-type UnionType declarations
+from __future__ import annotations
+
+from contextlib import contextmanager
+import logging
+from pathlib import Path
+from typing import 
+    Any,
+    Dict,
+    IO,
+)
+
```

Comment:
```suggestion

import datalad

from ..iter_collections.tarfile import TarfileItem

```

Reply:
It should not be TarfileItem below, but something more generic. Not yet sure in which way.

Reply:
> It should not be TarfileItem below, but something more generic. Not yet sure in which way.

I thought so too. I guess `FilesystemItem` would be more appropriate. First, for the name and second because `TarfileItem` does not actually extend `FilesystemItem`. We could rename it to something like `ArchiveItem`

Reply:
Changed it to FilesystemItem